node/src/accountant/scanners_tools.rs

```diff
@@ -0,0 +1,539 @@
+
+    pub fn timestamp_as_string(timestamp: &SystemTime) -> String {
+        let offset_date_time = OffsetDateTime::from(*timestamp);
+        offset_date_time
+            .format(&parse(TIME_FORMATTING_STRING).unwrap())
+            .unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::accountant::payable_dao::{Payable, PayableAccount};
+    use crate::accountant::receivable_dao::ReceivableAccount;
+    use crate::accountant::scanners_tools::payable_scanner_tools::{
+        calculate_payout_threshold, exceeded_summary, investigate_debt_extremes,
+        is_payable_qualified, payable_time_diff, qualified_payables_and_summary,
+        separate_early_errors,
+    };
+    use crate::accountant::scanners_tools::receivable_scanner_tools::balance_and_age;
+    use crate::accountant::SentPayable;
+    use crate::blockchain::blockchain_interface::BlockchainError;
+    use crate::database::dao_utils::{from_time_t, to_time_t};
+    use crate::sub_lib::accountant::PaymentThresholds;
+    use crate::test_utils::make_wallet;
+    use masq_lib::logger::Logger;
+    use std::rc::Rc;
+    use std::time::SystemTime;
+
+    #[test]
+    fn payable_generated_before_maturity_time_limit_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let qualified_debt = payment_thresholds.permanent_debt_allowed_gwei + 1;
+        let unqualified_time = to_time_t(now) - payment_thresholds.maturity_threshold_sec + 1;
+        let unqualified_payable_account = PayableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: qualified_debt,
+            last_paid_timestamp: from_time_t(unqualified_time),
+            pending_payable_opt: None,
+        };
+
+        let result = is_payable_qualified(now, &unqualified_payable_account, &payment_thresholds);
+
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn payable_with_low_debt_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let unqualified_debt = payment_thresholds.permanent_debt_allowed_gwei - 1;
+        let qualified_time = to_time_t(now) - payment_thresholds.maturity_threshold_sec - 1;
+        let unqualified_payable_account = PayableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: unqualified_debt,
+            last_paid_timestamp: from_time_t(qualified_time),
+            pending_payable_opt: None,
+        };
+
+        let result = is_payable_qualified(now, &unqualified_payable_account, &payment_thresholds);
+
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn payable_with_low_payout_threshold_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let debt = payment_thresholds.permanent_debt_allowed_gwei + 1;
```

Comment:
You should try to get as close to the threshold as possible, not to be almost as far as possible, the opposite... 
Try active computing of a triangle to get the slope of the declining line and then to compute a point somewhere on the threshold (ideally near to its middle) and then just subtract a few units (maybe just 1, 2, or 3...) to get a slightly smaller value. It will be as proofing as we can ever do.